component/otelcol/exporter/prometheus/internal/convert/convert.go

```diff
@@ -524,10 +563,44 @@ func (conv *Converter) consumeHistogram(app storage.Appender, memResource *memor
 			if err := writeSeries(app, infBucket, dp, infBucketVal); err != nil {
 				level.Error(conv.log).Log("msg", "failed to write histogram bucket sample", "bucket", bucketLabel.Value, "err", err)
 			}
+
+			// Add remaining exemplars.
+			for j := exemplarsIndex; j < len(exemplars); j++ {
+				if err := conv.writeExemplar(app, infBucket, exemplars[j]); err != nil {
+					level.Error(conv.log).Log("msg", "failed to add exemplar to bucket", bucketLabel.Value, "err", err)
+				}
+			}
 		}
 	}
 }
 
+// Convert Otel Exemplar to Prometheus Exemplar.
+func (conv *Converter) convertExemplar(otelExemplar pmetric.Exemplar, ts time.Time) exemplar.Exemplar {
+	exemplarLabels := make(labels.Labels, 0)
+
+	if traceID := otelExemplar.TraceID(); !traceID.IsEmpty() {
+		exemplarLabels = append(exemplarLabels, labels.Label{Name: "trace_id", Value: hex.EncodeToString(traceID[:])})
+	}
+
+	if spanID := otelExemplar.SpanID(); !spanID.IsEmpty() {
+		exemplarLabels = append(exemplarLabels, labels.Label{Name: "span_id", Value: hex.EncodeToString(spanID[:])})
+	}
+
+	var value float64
+	switch otelExemplar.ValueType() {
```

Comment:
Technically it can also be `ExemplarValueTypeEmpty` 🤔

Reply:
should we explicitly handle it? I took this function from the otel collector. It also says that "// Calling this function on zero-initialized Exemplar will cause a panic." on the ValueType function.